### PR TITLE
fix(capabilities): Generate default scope even without routes when ca…

### LINKED ISSUE #79: App without any routes generates no openapi.json despite having schemas
This broke with https://github.com/nextcloud/openapi-extractor/pull/24.
The comments app in server is affected, because it has no routes and only it's capabilities are present (deleted the openapi.json file, it will not be regenerated).

### PR DESCRIPTION
**Pull request by nickvergessen**
…pabilities exist

Fix #79

### CODE REVIEW

File: generate-spec [hunk 1]
```diff
@@ -778,6 +778,13 @@ $fullScopePathArrays = [];
 
 if (!$hasSingleScope) {
 	$scopePaths['full'] = [];
+} elseif (count($scopePaths) === 0) {
+	if (isset($schemas['Capabilities']) || isset($schemas['PublicCapabilities'])) {
+		Logger::info('app', 'Generating default scope without routes to populate capabilities');
+		$scopePaths['default'] = [];
+	} else {
+		Logger::panic('app', 'No routes or capabilities defined');
```
Comment (nickvergessen) [inline generate-spec:786]: Could also remove this, but I think it's helpful notifying that openapi is run uselessly

**Comment by provokateurin** [comment]
Testing this with the comments app still doesn't generate a new openapi.json

**Comment by nickvergessen** [comment]
> Testing this with the comments app still doesn't generate a new openapi.json

Well server does not regenerate if the file is absent. But if you add a complete empty `openapi.json` for comments and sharebymail both will be filled correctly.

https://github.com/nextcloud/server/blob/5a74f4ac398dda764d59952436bd7d36fbb84cf0/build/openapi-checker.sh#L3-L5

**Comment by nickvergessen** [comment]
That being said, something for later, but we should change server and just loop over all non-excluded directories and add a `.noopenapi` ignore file where we don't want to have it instead.
E.g. I noticed workflowengine and systemtags do/did not have it although providing capabilties and/or OCS endpoints

**Comment by provokateurin** [comment]
Yes, that's also relevant for https://github.com/nextcloud/openapi-extractor/pull/82 after which it is not given that a openapi.json file exists (without any scope suffix).

**Comment by provokateurin** [comment]
> Well server does not regenerate if the file is absent

My bad I thought I tested it correctly.

**Review by provokateurin — APPROVED** [review]

**Pull request merged**